src/smoothscroll.js

```diff
@@ -1,371 +1,436 @@
-(function(w, d, undefined) {
-  'use strict';
-
-  /*
-   * aliases
-   * w: window global object
-   * d: document
-   * undefined: undefined
+'use strict';
+
+/*
+ * aliases
+ * w: window global object
+ * d: document
+ */
+var w = window;
+var d = document;
+
+/**
+ * indicates if a the current browser is made by Microsoft
+ * @method isMicrosoftBrowser
+ * @param {String} userAgent
+ * @returns {Boolean}
+ */
+function isMicrosoftBrowser(userAgent) {
+  var userAgentPatterns = ['MSIE ', 'Trident/', 'Edge/'];
+
+  return new RegExp(userAgentPatterns.join('|')).test(userAgent);
+}
+
+ // polyfill
+function polyfill() {
+  // return if scroll behavior is supported and polyfill is not forced
+  if ('scrollBehavior' in d.documentElement.style
+    && w.__forceSmoothScrollPolyfill__ !== true) {
+    return;
+  }
+
+  // globals
+  var Element = w.HTMLElement || w.Element;
+  var ROUNDING_TOLERANCE = isMicrosoftBrowser(w.navigator.userAgent) ? 1 : 0;
+  var SCROLL_TIME = typeof w.__smoothScrollPolyfillSpeed__ === 'number'
```

Comment:
Yeah I have the same feeling, I encourage not to modify it in the README, I think that dynamic scroll time is **the next thing** to tackle here since the rest of the spec seems to be pretty much covered 😃

Reply:
The biggest thing here is if we expose it now we’re kinda forced to keep it exposed for a while for semver.

Reply:
Unless we release `major` which it's supposes to add breaking changes I think, right? If you consider this harmful I'm OK with deleting this.